TELCODOCS-597:  Adding a local registry to a Single Node OpenShift DU profile (#50904)
openshift/openshift-docs: modules/ztp-configuring-disk-partitioning.adoc

// Module included in the following assemblies:
//
// scalability_and_performance/ztp-deploying-disconnected.adoc

:_module-type: PROCEDURE
[id="ztp-configuring-disk-partitioning_{context}"]
= Configuring disk partitioning with SiteConfig

Use a `SiteConfig` CR to generate the `MachineConfig` CR used for disk partitioning. Prior to installation, you need to modify values in the `SiteConfig` CR to reflect dependencies on the underlying disk.

[NOTE]
====
You must use persistent naming for devices to avoid device names such as `/dev/sda` and `/dev/sdb` being switched at every reboot. You can use `rootDeviceHints` to choose the bootable device and then use same device for further partitioning: in this case, for Image registry.
====

.Prerequisites

* You have installed and configured Zero Touch Provisioning (ZTP). For information about this, see the topic on ZTP in Additional resources.

.Procedure

> . Add the following YAML to the `SiteConfig.yaml` file to generate `MachineConfig` for disk partitioning:

Reviewer: ```suggestion
. Add the following YAML to the `SiteConfig` CR that you use to generate the `MachineConfig` CR for disk partitioning:
```

Author: Done.

+
[source,yaml]
----
nodes:
  - rootDeviceHints:
      wwn: "0x62cea7f05c98c2002708a0a22ff480ea"
    diskPartition:
    - device: /dev/disk/by-id/wwn-0x62cea7f05c98c2002708a0a22ff480ea <1>
      partitions:
       - mount_point: /var/imageregistry
         size: 102500 <2>
         start: 344844 <3>
----
<1> This setting depends on the hardware. The setting can be a serial number or device name. The value must match the entry for `rootDeviceHint`.

> <2> The minimum value for image registry size is 100 gig (102500 MiB). For further information, see the Image Registry topic in Additional Resources.

Reviewer: ```suggestion
<2> The minimum value for `size` is 102500 MiB.
```

Author: Done.

> <3> The minimum value for the mount point is 25000 MiB. Otherwise the root partition is too small. Select `startMiB` (Disk - sizeMiB - some buffer). The total of startMiB + sizeMiB must not exceed the disk size, or the installation will fail.

Reviewer: ```suggestion
<3> The minimum value for `start` is 25000 MiB. The total value of `size` and `start` must not exceed the disk size, or the installation will fail.
```

Author: Done.

// Module included in the following assemblies:
//
// scalability_and_performance/ztp-deploying-disconnected.adoc

:_module-type: PROCEDURE
[id="ztp-configuring-disk-partitioning_{context}"]
= Configuring disk partitioning with SiteConfig

> You can use the `SiteConfig` YAML file to generate the `MachineConfig` CR used for disk partitioning. Prior to installation, you need to modify values in the `SiteConfig` CR to reflect dependencies on the underlying disk.

Reviewer: The CR is the resource that you patch/push to the hub cluster. The YAML file is just a representation of that resource. So the YAML files doesn't do anything on it's own, you apply it as a resource on the cluster to make the cluster updates. 

```suggestion
Use a `SiteConfig` CR to generate the `MachineConfig` CR used for disk partitioning. Prior to installation, you need to modify values in the `SiteConfig` CR to reflect dependencies on the underlying disk.
```

Author: Done.

[NOTE]
====
You must use persistent naming for devices to avoid device names such as `/dev/sda` and `/dev/sdb` being switched at every reboot. You can use `rootDeviceHints` to choose the bootable device and then use same device for further partitioning: in this case, for Image registry.
====

.Prerequisites

* You have installed and configured Zero Touch Provisioning (ZTP). For information about this, see the topic on ZTP in Additional resources.

.Procedure

. Add the following YAML to the `SiteConfig.yaml` file to generate `MachineConfig` for disk partitioning. :
+

> [source,yaml]

Reviewer: Suggest reviewing the updated notes in https://github.com/openshift/openshift-docs/pull/53083/files#diff-c0c2580378cc5e557211fa281f26dbe3efdd1da52e33bc7ef3873f9f10639307 and making appropriate changes here.

// Module included in the following assemblies:
//
// scalability_and_performance/ztp-deploying-disconnected.adoc

:_module-type: PROCEDURE
[id="ztp-configuring-disk-partitioning_{context}"]
= Configuring disk partitioning with SiteConfig

> You can use the `SiteConfig` YAML file to generate the `MachineConfig` CR used for disk partitioning. You need to to modify values in the `SiteConfig` CR to reflect dependencies on the underlying disk.

Reviewer 2: "to" is duplicated in the second sentence.

> The values must be established prior to installation.

Reviewer 2: This either needs an empty line above it, or it can be added to the statement above on line 9. 

Optionally, it seems high priority enough to warrant an `[IMPORTANT]` note, right? 

And what values are we referring to? `SiteConfig` or ``MachineConfig`?

Author: Went with "Prior to installation, you need to to modify values in the `SiteConfig` CR to reflect dependencies on the underlying disk."
MachineConfig is generated using SiteConfig.

[NOTE]
====
You must use persistent naming for devices to avoid device names such as `/dev/sda` and `/dev/sdb` being switched at every reboot. You can use `rootDeviceHints` to choose the bootable device and then use same device for further partitioning: in this case, for Image registry.
====

.Prerequisites

> 

Reviewer 2: Extra empty line

Author: Done.

> * You have installed and configured Zero Touch Provisioning (ZTP). For further information on this, see the topic on ZTP in Additional resources.

Reviewer 2: Use "information about..."

.Procedure

> . Add the following YAML to the `SiteConfig.yaml` file to generate `MachineConfig` for disk partitioning. :

Reviewer 2: disk partitioning:

Author: Done.

+
[source,yaml]
----
nodes:
  - rootDeviceHints:
      wwn: "0x62cea7f05c98c2002708a0a22ff480ea"
    diskPartition:
    - device: /dev/disk/by-id/wwn-0x62cea7f05c98c2002708a0a22ff480ea <1>
      partitions:
       - mount_point: /var/imageregistry
         size: 102500 <2>
         start: 344844 <3>
----
<1> This setting depends on the hardware. The setting can be a serial number or device name. The value must match the entry for `rootDeviceHint`.

> <2> The minimum value for image registry size is 100 gig (102500 MiB). For further information, see the Image Registry topic in Additional Resources.

Reviewer 2: I recommend matching format and switching the values for clarity:

"... size is 102500 MiB (100 GiB)."

Or just leave 100 GiB out: 

"The minimum value for image registry size is 102500 MiB."  

The above example would match the format of `<3>` below as well.

> <3> The minimum value for the mount point is 25000 MiB. Otherwise the root partition is too small. Select `startMiB` (Disk - sizeMiB - some buffer). The total of startMiB + sizeMiB must not exceed the disk size, or the installation will fail.

Reviewer 2: "Otherwise the root partition is too small" is an unnecessary sentence, given the previous sentence defines the minimum value. 

Is "(Disk - sizeMiB - some buffer)" a math equation like "startMiB + sizeMiB"? Please find another way to clearly state what is expected of the user.

// Module included in the following assemblies:
//
// scalability_and_performance/ztp-deploying-disconnected.adoc

:_module-type: PROCEDURE
[id="ztp-configuring-disk-partitioning_{context}"]
= Configuring disk partitioning with SiteConfig

> You can use the `SiteConfig` YAML file to generate the `MachineConfig` CR used for disk partitioning. You need to to modify values in the `MachineConfig` CR to reflect dependencies on the underlying disk.

Reviewer 3: `You need to to modify values in the MachineConfig CR to reflect dependencies on the underlying disk.`  - imo this statement is not clear. The user entrypoint is the SiteConfig and that's where the disk related values are added.  The user doesn't edit MachineConfig directly as they get generated from the SiteConfig.

Author: Done.

The values must be established prior to installation.

[NOTE]
====
You must use persistent naming for devices to avoid device names such as `/dev/sda` and `/dev/sdb` being switched at every reboot. You can use `rootDeviceHints` to choose the bootable device and then use same device for further partitioning: in this case, for Image registry.
====

.Prerequisites

* You have installed and configured Zero Touch Provisioning (ZTP). For further information on this, see the topic on ZTP in Additional resources.

.Procedure

. Add the following YAML to the `SiteConfig.yaml` file to generate `MachineConfig` for disk partitioning. :
+
[source,yaml]
----

> #Comment ....

Reviewer 4: remove this line

Author: Done.

nodes:
  - rootDeviceHints:
      wwn: "0x62cea7f05c98c2002708a0a22ff480ea"
    diskPartition:
    - device: /dev/disk/by-id/wwn-0x62cea7f05c98c2002708a0a22ff480ea <1>
      partitions:
       - mount_point: /var/imageregistry
         size: 102500 <2>
         start: 344844 <3>
----

> <1> Depends on the hardware. The setting can also be a serial number or device name, for example `wwn`. The value must match with the `win` entry for `rootDeviceHint`.

Reviewer 4: 1 - the example is already using wwn (remove "for example `wwn`")
2- "The value must match with the `win` entry for `rootDeviceHint`" change to "The value must match with the entry for `rootDeviceHint`" (drop 'win')

Author: Done.
